From 61f3c2299a83c8c21dc4de64f3616e8ed967714c Mon Sep 17 00:00:00 2001
From: Naveen Mamindlapalli <naveenm@marvell.com>
Date: Thu, 8 Apr 2021 13:24:26 +0530
Subject: [PATCH 762/767] octeontx2-bphy-netdev: Fix common lmac resource free

commit 5d30bd71435ea533afe304b8a2685aa26f5fb63d from
git@git.assembla.com:cavium/WindRiver.linux.git

This patch fixes multiple instances of kfree() of rfoe_common
and cpri_common objects shared by RFOE LMAC interfaces and CPRI
LMAC interfaces.

Change-Id: I2c64a26a35d2ea4369be39c3cca5075a499953f9
Signed-off-by: Naveen Mamindlapalli <naveenm@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49736
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <SunilKovvuri.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c | 2 ++
 drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c
index 95e78df0ab9f..11f411adc194 100644
--- a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c
+++ b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c
@@ -106,6 +106,7 @@ void otx2_bphy_cpri_cleanup(void)
 			unregister_netdev(netdev);
 			netif_napi_del(&priv->napi);
 			kfree(priv->cpri_common);
+			priv->cpri_common = NULL;
 			free_netdev(netdev);
 			drv_ctx->valid = 0;
 		}
@@ -621,6 +622,7 @@ int otx2_cpri_parse_and_init_intf(struct otx2_bphy_cdev_priv *cdev,
 			unregister_netdev(netdev);
 			netif_napi_del(&priv->napi);
 			kfree(priv->cpri_common);
+			priv->cpri_common = NULL;
 			free_netdev(netdev);
 			drv_ctx->valid = 0;
 		}
diff --git a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c
index 38d1a3e5af06..378d7d7514a3 100644
--- a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c
+++ b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c
@@ -144,6 +144,7 @@ void otx2_bphy_rfoe_cleanup(void)
 				netif_napi_del(&ft_cfg->napi);
 			}
 			kfree(priv->rfoe_common);
+			priv->rfoe_common = NULL;
 			free_netdev(netdev);
 			drv_ctx->valid = 0;
 		}
@@ -1377,6 +1378,7 @@ int otx2_rfoe_parse_and_init_intf(struct otx2_bphy_cdev_priv *cdev,
 				netif_napi_del(&ft_cfg->napi);
 			}
 			kfree(priv->rfoe_common);
+			priv->rfoe_common = NULL;
 			free_netdev(netdev);
 			drv_ctx->valid = 0;
 		}
-- 
2.31.1

